extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1236,6 +1250,59 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       (returnedSizeDifferent && heightNotIncreased)
     ) {
       this.attemptChangeSize(newHeight, newWidth).catch(() => {});
+      if (newWidth > width) {
+        this.adjustSlotPostExpansion_(newWidth);
+      }
+    }
+  }
+
+  /**
+   * Ensures that slot is properly centered after being expanded.
+   * @param {number} newWidth The new width of the slot.
+   * @private
+   */
+  adjustSlotPostExpansion_(newWidth) {
+    if (this.flexibleAdSlotData_ == null) {
+      dev().warn(
+        TAG,
+        'Attempted to expand slot without flexible ad slot data.'
+      );
+      return;
+    }
+    const isRtl = isRTL(this.win.document);
+    const dirStr = isRtl ? 'Right' : 'Left';
+    // Guaranteed to be set after exiting if/else.
+    let /** ?number */ newMargin = null;
```

Comment:
Might as well expand portions of this.flexibleAdSlotData_ here:

const {parentWidth, parentStyle} = this.flexibleAdSlotData_;

Reply:
Done.